spec/lang/well-formed.md

```diff
@@ -213,49 +228,55 @@ impl ValueExpr {
 
                 match t {
                     Type::Tuple { fields, .. } => {
-                        ensure(exprs.len() == fields.len())?;
+                        ensure_wf(exprs.len() == fields.len(), "ValueExpr::Tuple: invalid number of fields")?;
                         for (e, (_offset, ty)) in exprs.zip(fields) {
                             let checked = e.check_wf::<T>(locals, prog)?;
-                            ensure(checked == ty)?;
+                            ensure_wf(checked == ty, "ValueExpr::Tuple: invalid field type")?;
                         }
                     },
                     Type::Array { elem, count } => {
-                        ensure(exprs.len() == count)?;
+                        ensure_wf(exprs.len() == count, "ValueExpr::Tuple: invalid number of elements")?;
                         for e in exprs {
                             let checked = e.check_wf::<T>(locals, prog)?;
-                            ensure(checked == elem)?;
+                            ensure_wf(checked == elem, "ValueExpr::Tuple::Array: invalid element type")?;
```

Comment:
```suggestion
                            ensure_wf(checked == elem, "ValueExpr::Array: invalid element type")?;
```
`Tuple` seems wrong?

Reply:
Ah, good point. What about "ValueExpr::Tuple: invalid array element type"?
And then below, `"ValueExpr::Tuple: invalid number of array elements"?

Reply:
That works fine I think. :)